homeassistant/components/velbus/config_flow.py

```diff
@@ -144,3 +163,74 @@ async def async_step_discovery_confirm(
             step_id="discovery_confirm",
             description_placeholders={CONF_NAME: self._title},
         )
+
+    async def _validate_vlp_file(self, file_path: str) -> None:
+        """Validate VLP file and raise exception if invalid."""
+        vlpfile = VlpFile(file_path)
+        await vlpfile.read()
+        if len(vlpfile.get()) == 0:
+            raise InvalidVlpFile("no_modules")
+
+    async def async_step_vlp(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Step when user wants to use the VLP file."""
+        if user_input is not None:
+            if CONF_VLP_FILE not in user_input or user_input[CONF_VLP_FILE] == "":
+                # The VLP file is optional, so allow skipping it
+                self._vlp_file = ""
+            else:
+                try:
+                    # handle the file upload
+                    self._vlp_file = await self.hass.async_add_executor_job(
+                        save_uploaded_vlp_file, self.hass, user_input[CONF_VLP_FILE]
+                    )
+                    # validate it
+                    await self._validate_vlp_file(self._vlp_file)
+                except InvalidVlpFile as e:
+                    self._errors[CONF_VLP_FILE] = str(e)
+            if self.source == SOURCE_RECONFIGURE:
+                old_entry = self._get_reconfigure_entry()
+                return self.async_update_reload_and_abort(
+                    old_entry,
+                    data={
+                        CONF_VLP_FILE: self._vlp_file,
+                        CONF_PORT: old_entry.data.get(CONF_PORT, None),
```

Comment:
Can the port be missing in the data?
```suggestion
                        CONF_PORT: old_entry.data.get(CONF_PORT),
```

Reply:
without port the integration can not work, it has always been a required parameter